mcs/class/System/System.Net/ResponseStream.cs

```diff
@@ -81,19 +81,25 @@ public override void Close ()
 				byte [] bytes = null;
 				MemoryStream ms = GetHeaders (true);
 				bool chunked = response.SendChunked;
-				if (ms != null) {
-					long start = ms.Position;
-					if (chunked && !trailer_sent) {
-						bytes = GetChunkSizeBytes (0, true);
-						ms.Position = ms.Length;
-						ms.Write (bytes, 0, bytes.Length);
+				if (stream.CanWrite) {
+					try {
+						if (ms != null) {
+							long start = ms.Position;
+							if (chunked && !trailer_sent) {
+								bytes = GetChunkSizeBytes (0, true);
+								ms.Position = ms.Length;
+								ms.Write (bytes, 0, bytes.Length);
+							}
+							InternalWrite (ms.GetBuffer (), (int) start, (int) (ms.Length - start));
+							trailer_sent = true;
+						} else if (chunked && !trailer_sent) {
+							bytes = GetChunkSizeBytes (0, true);
+							InternalWrite (bytes, 0, bytes.Length);
+							trailer_sent = true;
+						}
+					} catch (IOException ex) {
```

Comment:
Swallowing exceptions is almost always a terrible idea, are there any IOException beyond connection reset by peer that we should be reporting?

And shouldn't that exception be visible anyways?